config/network/errors.go

```diff
@@ -0,0 +1,30 @@
+package network
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/onflow/flow-go/network/p2p"
+)
+
+// ErrInvalidLimitConfig indicates the validation limit is < 0.
+type ErrInvalidLimitConfig struct {
+	// controlMsg the control message type.
+	controlMsg p2p.ControlMessageType
+	err        error
+}
+
+func (e ErrInvalidLimitConfig) Error() string {
+	return fmt.Errorf("invalid rpc control message %s validation limit configuration: %w", e.controlMsg, e.err).Error()
+}
+
+// NewInvalidLimitConfigErr returns a new ErrValidationLimit.
+func NewInvalidLimitConfigErr(controlMsg p2p.ControlMessageType, err error) ErrInvalidLimitConfig {
+	return ErrInvalidLimitConfig{controlMsg: controlMsg, err: err}
```

Comment:
```suggestion
func NewInvalidLimitConfigError(controlMsg p2p.ControlMessageType, msg string, args ...any) ErrInvalidLimitConfig {
	return InvalidLimitConfigError{controlMsg: controlMsg, err: fmt.Errorf(msg, args...)}
```

Optional: I find this is a useful pattern to define error types:
- guarantees `MyError.err` is non-nil
- invocation is similar to `fmt.Errorf` + don't need to embed `fmt.Errorf` at the callsite + encourages the caller to add additional context (like `Errorf`)

Reply:
This pattern is less useful for the usage of NewInvalidLimitConfigError since the error is simple the only context needed is the control msg type and the validation err to wrap.